clientdb/batch_snapshot.go

```diff
@@ -481,6 +504,34 @@ func deserializeLocalBatchSnapshot(r io.Reader) (*LocalBatchSnapshot, error) {
 		b.MatchedOrders[nonce] = append(b.MatchedOrders[nonce], m)
 	}
 
+	// Older batches had a single clearing price instead of a map. If we
+	// have a non-zero price, it means this is an old snapshot and we don't
+	// need to read any further.
+	if clearingPrice > 0 {
+		b.ClearingPrices[order.LegacyLeaseDurationBucket] = clearingPrice
+
+		return b, nil
+	}
+
+	var (
+		numPrices uint32
+		duration  uint32
+		price     order.FixedRatePremium
+	)
+	err = ReadElements(r, &numPrices)
+	if err != nil {
+		return nil, err
+	}
+
+	for i := uint32(0); i < numPrices; i++ {
+		err = ReadElements(r, &duration, &price)
```

Comment:
this is probably okay since these are just `uint32`, but triggers my spidey sense :P Could define these in the inner scope to make clear that each iteration is not overwriting the previous value.